api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -453,6 +458,34 @@ protected Type.TypeID typeId() {
     }
   }
 
+  static class TimestampNanoLiteral extends ComparableLiteral<Long> {
+    TimestampNanoLiteral(Long value) {
+      super(value);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Literal<T> to(Type type) {
+      switch (type.typeId()) {
+        case DATE:
+          return (Literal<T>)
+              new DateLiteral(
+                  (int) ChronoUnit.DAYS.between(EPOCH_DAY, EPOCH.plusNanos(value()).toLocalDate()));
```

Comment:
Is there a DateTimeUtil` method for this? Seems like it would be better to colocate the logic there.

Reply:
Fixed here: https://github.com/jacobmarble/apache-iceberg/pull/1